presentation/src/main/java/com/dcs/presentation/ui/home/HomeScreen.kt

```diff
@@ -1,27 +1,249 @@
+import androidx.compose.runtime.remember
+import androidx.compose.runtime.setValue
 import androidx.compose.ui.Modifier
+import androidx.compose.ui.platform.LocalFocusManager
+import androidx.compose.ui.platform.LocalView
+import androidx.compose.ui.res.stringResource
+import androidx.compose.ui.text.font.FontWeight
+import androidx.compose.ui.text.style.TextAlign
+import androidx.compose.ui.tooling.preview.Preview
+import androidx.compose.ui.tooling.preview.PreviewParameter
+import androidx.compose.ui.unit.dp
+import androidx.core.view.ViewCompat
+import androidx.core.view.WindowInsetsCompat
 import androidx.hilt.navigation.compose.hiltViewModel
+import androidx.lifecycle.compose.collectAsStateWithLifecycle
+import androidx.navigation.NavHostController
+import com.dcs.presentation.R
+import com.dcs.presentation.core.model.KeywordUiState
+import com.dcs.presentation.core.model.KeywordUiStateProvider
+import com.dcs.presentation.core.theme.AllAboutMovieTheme
+import com.dcs.presentation.ui.Screen
 
 @Composable
 fun HomeRoute(
+    navController: NavHostController,
+    viewModel: HomeViewModel = hiltViewModel(),
+) {
+    val searchHistory by viewModel.searchHistory.collectAsStateWithLifecycle()
+
+    LaunchedEffect(key1 = HomeViewModel::class) {
+        viewModel.effect.collect {
+            when (it) {
+                HomeEffect.NavigateBack -> {
+                    navController.popBackStack()
+                }
+
+                is HomeEffect.NavigateToSearchResult -> {
+                    navController.navigate(Screen.SearchResult.createRoute(it.keyword))
+                }
+            }
+        }
+    }
+
+    HomeScreen(
+        queryText = viewModel.queryText,
+        searchHistory = searchHistory,
+        onHomeUiEvent = viewModel::dispatchEvent,
+        modifier = Modifier.fillMaxSize()
+    )
+}
+
+@Composable
+private fun HomeScreen(
+    queryText: String,
+    searchHistory: List<KeywordUiState>,
+    onHomeUiEvent: (HomeUiEvent) -> Unit,
     modifier: Modifier = Modifier,
-    viewModel: HomeViewModel = hiltViewModel()
 ) {
+    Row(
+        modifier = modifier,
+    ) {
+        SearchTopBar(
+            queryText = queryText,
+            searchHistory = searchHistory,
+            onHomeUiEvent = onHomeUiEvent,
+        )
+    }
+}
+
+@OptIn(ExperimentalMaterial3Api::class)
+@Composable
+private fun SearchTopBar(
+    queryText: String,
+    searchHistory: List<KeywordUiState>,
+    onHomeUiEvent: (HomeUiEvent) -> Unit,
+) {
+    val focusManager = LocalFocusManager.current
+    val view = LocalView.current
+    val viewTreeObserver = view.viewTreeObserver
+    DisposableEffect(viewTreeObserver) {
+        val listener = ViewTreeObserver.OnGlobalLayoutListener {
+            val isKeyboardOpen = ViewCompat.getRootWindowInsets(view)
+                ?.isVisible(WindowInsetsCompat.Type.ime()) ?: true
+
+            if (!isKeyboardOpen) {
+                focusManager.clearFocus()
+            }
+        }
```

Comment:
```suggestion
    DisposableEffect(viewTreeObserver, active) {
        val listener = ViewTreeObserver.OnGlobalLayoutListener {
            val isKeyboardOpen = ViewCompat.getRootWindowInsets(view)
                ?.isVisible(WindowInsetsCompat.Type.ime()) ?: true

            if (!isKeyboardOpen) {
                if (active) active = false
                focusManager.clearFocus()
            }
        }
```
이런식으로 하면 자연스럽게 키보드 내려가면서 검색바 화면 내려 갈 것 같아요. 제가 active statehoisting 얘기 하면서 남긴 코멘트 코드 참고 해주세요!

Reply:
우리 시나리오는 투스텝.